lang/cs/Org.Apache.REEF.IMRU.Tests/TestTaskManager.cs

```diff
@@ -0,0 +1,679 @@
+        /// Tests adding all mapper tasks without master task
+        /// </summary>
+        [Fact]
+        public void NoMasterTask()
+        {
+            var taskManager = CreateTaskManager();
+            taskManager.AddTask(MapperTaskIdPrefix + 1, MockConfig(), CreateMockActiveContext(1));
+            taskManager.AddTask(MapperTaskIdPrefix + 2, MockConfig(), CreateMockActiveContext(1));
+            Action add = () => taskManager.AddTask(MapperTaskIdPrefix + 3, MockConfig(), CreateMockActiveContext(3));
+            Assert.Throws<IMRUSystemException>(add);
+        }
+
+        /// <summary>
+        /// Tests adding more than expected tasks
+        /// </summary>
+        [Fact]
+        public void ExceededTotalNumber()
+        {
+            var taskManager = TaskManagerWithTasksAdded();
+            Action add = () => taskManager.AddTask(MapperTaskIdPrefix + 4, MockConfig(), CreateMockActiveContext(4));
+            Assert.Throws<IMRUSystemException>(add);
+        }
+
+        /// <summary>
+        /// Tests adding a task with duplicated task id and duplicated master id
+        /// </summary>
+        [Fact]
+        public void DuplicatedTaskIdInAdd()
+        {
+            var taskManager = CreateTaskManager();
+            taskManager.AddTask(MasterTaskId, MockConfig(), CreateMockActiveContext(0));
+            taskManager.AddTask(MapperTaskIdPrefix + 1, MockConfig(), CreateMockActiveContext(1));
+            Action add = () => taskManager.AddTask(MapperTaskIdPrefix + 1, MockConfig(), CreateMockActiveContext(1));
+            Assert.Throws<IMRUSystemException>(add);
+            add = () => taskManager.AddTask(MasterTaskId, MockConfig(), CreateMockActiveContext(1));
+            Assert.Throws<IMRUSystemException>(add);
+        }
+
+        /// <summary>
+        /// Tests invalid arguments when adding tasks
+        /// </summary>
+        [Fact]
+        public void NullArguments()
+        {
+            var taskManager = CreateTaskManager();
+            taskManager.AddTask(MasterTaskId, MockConfig(), CreateMockActiveContext(0));
+
+            Action add = () => taskManager.AddTask(MapperTaskIdPrefix + 1, null, CreateMockActiveContext(1));
+            Assert.Throws<IMRUSystemException>(add);
+
+            add = () => taskManager.AddTask(MapperTaskIdPrefix + 2, MockConfig(), null);
+            Assert.Throws<IMRUSystemException>(add);
+        }
+
+        /// <summary>
+        /// Tests passing invalid arguments in creating TaskManager
+        /// </summary>
+        [Fact]
+        public void InvalidArgumentsInCreatingTaskManger()
+        {
+            Action taskManager = () => CreateTaskManager(0, MasterTaskId);
+            Assert.Throws<IMRUSystemException>(taskManager);
+
+            taskManager = () => CreateTaskManager(1, null);
+            Assert.Throws<IMRUSystemException>(taskManager);
+        }
+
+        /// <summary>
+        /// Tests set tasks to complete for running tasks
```

Comment:
Replace with "Tests whether all tasks rightly reach Running and Completed states"